CI fails to install cryptography 3.4.1

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### Information

- **Qiskit Aqua version**: NA
- **Python version**: NA
- **Operating system**: NA

### What is the current behavior?

CI fails to install cryptography 3.4.1.
If we update pip, it works. But, the current CI config seems to force to use pip 9 as follows.
https://github.com/Qiskit/qiskit-aqua/blob/7c2c2e546e9002e758ff6b25f98a9a594589bd6f/.github/actions/install-libraries/action.yml#L24

https://github.com/Qiskit/qiskit-aqua/pull/1527/checks?check_run_id=1852012104

```
Collecting cryptography>=1.3 (from requests-ntlm>=1.1.0->qiskit-ibmq-provider->-r requirements-dev.txt (line 18))
  Downloading https://files.pythonhosted.org/packages/06/ed/cb79cc94ec58d9d92557238fc6c629cd6e07d72334d2de556aecc2211370/cryptography-3.4.1.tar.gz (544kB)
    Complete output from command python setup.py egg_info:
    
            =============================DEBUG ASSISTANCE==========================
            If you are seeing an error here please try the following to
            successfully install cryptography:
    
            Upgrade to the latest pip and try again. This will fix errors for most
            users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
            =============================DEBUG ASSISTANCE==========================
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-etw2xocn/cryptography/setup.py", line 14, in <module>
        from setuptools_rust import RustExtension
    ModuleNotFoundError: No module named 'setuptools_rust'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-etw2xocn/cryptography/
You are using pip version 9.0.3, however version 21.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Error: Process completed with exit code 1.
```


### Steps to reproduce the problem




### What is the expected behavior?



### Suggested solutions

--- a/.github/actions/install-master-dependencies/action.yml
+++ b/.github/actions/install-master-dependencies/action.yml
@@ -23,6 +23,5 @@ runs:
         fi
         pip install https://github.com/Qiskit/qiskit-terra/archive/master.zip
         pip install https://github.com/Qiskit/qiskit-ignis/archive/master.zip
-        pip install https://github.com/Qiskit/qiskit-ibmq-provider/archive/master.zip
         pip install https://github.com/Qiskit/qiskit-aer/archive/master.zip
       shell: bash

--- a/qiskit/aqua/operators/list_ops/tensored_op.py
+++ b/qiskit/aqua/operators/list_ops/tensored_op.py
@@ -20,7 +20,6 @@ from qiskit.circuit import QuantumCircuit, ParameterExpression
 
 from ..state_fns.circuit_state_fn import CircuitStateFn
 from ..operator_base import OperatorBase
-from ..primitive_ops.primitive_op import PrimitiveOp
 from .list_op import ListOp
 from ... import AquaError
 
@@ -97,6 +96,7 @@ class TensoredOp(ListOp):
             AquaError: for operators where a single underlying circuit can not be produced.
         """
         circuit_op = self.to_circuit_op()
+        from ..primitive_ops.primitive_op import PrimitiveOp
         if isinstance(circuit_op, (PrimitiveOp, CircuitStateFn)):
             return circuit_op.to_circuit()
         raise AquaError('Conversion to_circuit supported only for operators, where a single '

--- a/qiskit/chemistry/qmolecule.py
+++ b/qiskit/chemistry/qmolecule.py
@@ -1,6 +1,6 @@
 # This code is part of Qiskit.
 #
-# (C) Copyright IBM 2018, 2020.
+# (C) Copyright IBM 2018, 2021.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -191,7 +191,7 @@ class QMolecule:
             with h5py.File(self._filename, "r") as file:
                 def read_array(name):
                     _data = file[name][...]
-                    if _data.dtype == numpy.bool and _data.size == 1 and not _data:
+                    if _data.dtype == numpy.bool_ and _data.size == 1 and not _data:
                         _data = None
                     return _data
 

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -15,7 +15,6 @@ jupyter-sphinx
 discover
 torch; sys_platform == 'linux' or (python_version < '3.8' and sys_platform != 'win32')
 qiskit-aer
-qiskit-ibmq-provider
 mypy>=0.780
 mypy-extensions>=0.4.3
 networkx>=2.2

--- a/test/chemistry/test_readme_sample.py
+++ b/test/chemistry/test_readme_sample.py
@@ -1,6 +1,6 @@
 # This code is part of Qiskit.
 #
-# (C) Copyright IBM 2020.
+# (C) Copyright IBM 2020, 2021.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -38,7 +38,8 @@ class TestReadmeSample(QiskitChemistryTestCase):
             # pylint: disable=import-outside-toplevel
             # pylint: disable=unused-import
             from qiskit import Aer
-        except Exception as ex:  # pylint: disable=broad-except
+            _ = Aer.get_backend('statevector_simulator')
+        except ImportError as ex:  # pylint: disable=broad-except
             self.skipTest("Aer doesn't appear to be installed. Error: '{}'".format(str(ex)))
             return
 

--- a/test/chemistry/test_uccsd_hartree_fock.py
+++ b/test/chemistry/test_uccsd_hartree_fock.py
@@ -1,6 +1,6 @@
 # This code is part of Qiskit.
 #
-# (C) Copyright IBM 2019, 2020.
+# (C) Copyright IBM 2019, 2021.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -91,10 +91,10 @@ class TestUCCSDHartreeFock(QiskitChemistryTestCase):
         try:
             # pylint: disable=import-outside-toplevel
             from qiskit import Aer
-        except Exception as ex:  # pylint: disable=broad-except
+            backend = Aer.get_backend('statevector_simulator')
+        except ImportError as ex:  # pylint: disable=broad-except
             self.skipTest("Aer doesn't appear to be installed. Error: '{}'".format(str(ex)))
             return
-        backend = Aer.get_backend('statevector_simulator')
         solver = VQE(var_form=self.var_form, optimizer=self.optimizer,
                      quantum_instance=QuantumInstance(backend=backend))
 
@@ -108,10 +108,10 @@ class TestUCCSDHartreeFock(QiskitChemistryTestCase):
         try:
             # pylint: disable=import-outside-toplevel
             from qiskit import Aer
-        except Exception as ex:  # pylint: disable=broad-except
+            backend = Aer.get_backend('qasm_simulator')
+        except ImportError as ex:  # pylint: disable=broad-except
             self.skipTest("Aer doesn't appear to be installed. Error: '{}'".format(str(ex)))
             return
-        backend = Aer.get_backend('qasm_simulator')
         optimizer = SPSA(maxiter=200, last_avg=5)
         solver = VQE(var_form=self.var_form, optimizer=optimizer,
                      expectation=PauliExpectation(),
@@ -129,10 +129,10 @@ class TestUCCSDHartreeFock(QiskitChemistryTestCase):
         try:
             # pylint: disable=import-outside-toplevel
             from qiskit import Aer
-        except Exception as ex:  # pylint: disable=broad-except
+            backend = Aer.get_backend('qasm_simulator')
+        except ImportError as ex:  # pylint: disable=broad-except
             self.skipTest("Aer doesn't appear to be installed. Error: '{}'".format(str(ex)))
             return
-        backend = Aer.get_backend('qasm_simulator')
         optimizer = SPSA(maxiter=200, last_avg=5)
         solver = VQE(var_form=self.var_form, optimizer=optimizer,
                      expectation=AerPauliExpectation(),

--- a/test/chemistry/test_vqe_uccsd_adapt.py
+++ b/test/chemistry/test_vqe_uccsd_adapt.py
@@ -1,6 +1,6 @@
 # This code is part of Qiskit.
 #
-# (C) Copyright IBM 2018, 2020.
+# (C) Copyright IBM 2018, 2021.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -71,13 +71,13 @@ class TestVQEAdaptUCCSD(QiskitChemistryTestCase):
         try:
             # pylint: disable=import-outside-toplevel
             from qiskit import Aer
-        except Exception as ex:  # pylint: disable=broad-except
+            backend = Aer.get_backend('statevector_simulator')
+        except ImportError as ex:  # pylint: disable=broad-except
             self.skipTest("Aer doesn't appear to be installed. Error: '{}'".format(str(ex)))
             return
 
         self.var_form_base = UCCSD(self.num_spin_orbitals,
                                    self.num_particles, initial_state=self.init_state)
-        backend = Aer.get_backend('statevector_simulator')
         optimizer = L_BFGS_B()
 
         warnings.filterwarnings('ignore', category=DeprecationWarning)

--- a/tox.ini
+++ b/tox.ini
@@ -14,7 +14,6 @@ setenv =
 deps = https://github.com/Qiskit/qiskit-terra/archive/master.zip
        https://github.com/Qiskit/qiskit-ignis/archive/master.zip
        https://github.com/Qiskit/qiskit-aer/archive/master.zip
-       https://github.com/Qiskit/qiskit-ibmq-provider/archive/master.zip
        -r{toxinidir}/requirements-dev.txt
 commands =
   stestr run {posargs}

